sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/SafeDivideOperatorConversion.java

```diff
@@ -33,9 +34,10 @@ public class SafeDivideOperatorConversion extends DirectOperatorConversion
 {
   private static final SqlFunction SQL_FUNCTION = OperatorConversions
       .operatorBuilder(StringUtils.toUpperCase(Function.SafeDivide.NAME))
-      .operandTypeChecker(OperandTypes.ANY_NUMERIC)
```

Comment:
Refactored this slightly and added the proper return type inference. Quotient depended on if the numerator and denominators are null and would throw an exception of failing datatype `EXPR$0: INTEGER NOT NULL -> INTEGER`